### DISCUSSION TITLE
Lesson 4 Challenge Help

### DISCUSSION THREAD

**Discussion by matthewg49** [Cyfrin/foundry-full-course-cu#163] [Q&A] [answered]
Hey all,

I'm having a bit of trouble understanding my solution for the challenge. I was able to find the correct answer
<details>
<summary><h3>SPOILER ALERT</h3></summary>
I grabbed the numerical value from the price feed and threw that into a gwei -> wei calculator here https://www.alchemy.com/gwei-calculator

But I noticed that it was only 21 digits long and notice the other answers were 23 digits long, so I added two zeroes and that worked. When I threw that number into the wei category in the calculator above, it seems to be one decimal point larger than the actual price of eth in USD. 

For example
* Here is my transaction https://arbiscan.io/tx/0x8e49dd4d0bb318d6ebf85e46218ca5e3496f9214006c869d2a4004acdd6d9d2f
* The answer in wei is 18787150000000000000000 and the answer in "ether" (technically usd in this case I believe) is 18787.15 which looks to be one decimal place too large, otherwise that means eth is $18k and I can go celebrate.

I'm confused by the larger numbers and when they need more zeroes 😅 
</details>

Could anyone help me figure out how to wrap my head around this?

**Comment by matthewg49** [comment, marked as answer]
Wait!! I think I know why now. Although the $ may not be "correct" the calculation in the smart contract changes the "actual price" in such a way that someone can't just look up the price in coingecko and throw it in as the answer. 

Lmk if I have that right :)

**Reply by PatrickAlphaC** [reply]
Well if you minted it you'd have it right ;)

**Reply by matthewg49** [reply]
NFT secured 🫡
